File: Ser2025/Company/Company.Web/Modules/Company/BusinessEntityAddress/BusinessEntityAddressForm.cs
using Serenity.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.BusinessEntityAddress")]
[BasedOnRow(typeof(BusinessEntityAddressRow), CheckNames = true)]
public class BusinessEntityAddressForm
{
    public int AddressId { get; set; }
    public int AddressTypeId { get; set; }
    public short IsActive { get; set; }
    public int TenantId { get; set; }
}

File: Ser2025/Company/Company.Web/Modules/ServerTypes/Company/BusinessEntityAddressForm.ts
import { PrefixedContext, IntegerEditor, initFormType } from '@serenity-is/corelib';

export interface BusinessEntityAddressForm {
    AddressId: IntegerEditor;
    AddressTypeId: IntegerEditor;
    IsActive: IntegerEditor;
    TenantId: IntegerEditor;
}

export class BusinessEntityAddressForm extends PrefixedContext {
    static readonly formKey = 'Company.BusinessEntityAddress';
    private static init: boolean;
    
    constructor(prefix: string) {
        super(prefix);
        if (!BusinessEntityAddressForm.init)  {
            BusinessEntityAddressForm.init = true;
            
            var w0 = IntegerEditor;

            initFormType(BusinessEntityAddressForm, [
            'AddressId', w0,
            'AddressTypeId', w0,
            'IsActive', w0,
            'TenantId', w0,
            ]);
        }
    }
}